tsdb/compact.go

```diff
@@ -656,7 +656,9 @@ func (c *LeveledCompactor) populateBlock(blocks []BlockReader, meta *BlockMeta,
 	defer func() {
 		var merr tsdb_errors.MultiError
 		merr.Add(err)
-		merr.Add(closeAll(closers))
+		if cerr := closeAll(closers); cerr != nil {
+			merr.Add(errors.Wrap(err, "close"))
```

Comment:
Oops, sorry for not noticing before
```suggestion
			merr.Add(errors.Wrap(eerr, "close"))
```

Reply:
Reason why `merr.Add(closeAll(closers))` is a less error prone approach. But let's not start a flame 🔥  😄

Reply:
xD

Reply:
BTW there should be vet for this

Reply:
Also I cared about wrapping mostly here